opm/material/densead/EvaluationV.hpp

```diff
@@ -0,0 +1,602 @@
+    { return value() >= other.value(); }
+
+    template <class RhsValueType>
+    bool operator<=(RhsValueType other) const
+    { return value() <= other; }
+
+    bool operator<=(const EvaluationV& other) const
+    { return value() <= other.value(); }
+
+    // return value of variable
+    const ValueType& value() const
+    { return data_[valuepos_]; }
+
+    // set value of variable
+    template <class RhsValueType>
+    void setValue(const RhsValueType& val)
+    { data_[valuepos_] = val; }
+
+    // return varIdx'th derivative
+    const ValueType& derivative(const int varIdx) const
+    {
+        assert(0 <= varIdx && varIdx < size);
+
+        return data_[dstart_ + varIdx];
+    }
+
+    // set derivative at position varIdx
+    void setDerivative(const int varIdx, const ValueType& derVal)
+    {
+        assert(0 <= varIdx && varIdx < size);
+
+        data_[dstart_ + varIdx] = derVal;
+    }
+
+private:
+    std::vector<ValueT> data_;
+};
+
+// the generic operators are only required for the unspecialized case
+template <class RhsValueType, class ValueType>
+bool operator<(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{ return b > a; }
+
+template <class RhsValueType, class ValueType>
+bool operator>(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{ return b < a; }
+
+template <class RhsValueType, class ValueType>
+bool operator<=(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{ return b >= a; }
+
+template <class RhsValueType, class ValueType>
+bool operator>=(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{ return b <= a; }
+
+template <class RhsValueType, class ValueType>
+bool operator!=(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{ return a != b.value(); }
+
+template <class RhsValueType, class ValueType>
+EvaluationV<ValueType> operator+(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{
+    EvaluationV<ValueType> result(b);
+    result += a;
+    return result;
+}
+
+template <class RhsValueType, class ValueType>
+EvaluationV<ValueType> operator-(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{
+    EvaluationV<ValueType> result(a);
+    result -= b;
+    return result;
+}
+
+template <class RhsValueType, class ValueType>
+EvaluationV<ValueType> operator/(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{
+    EvaluationV<ValueType> tmp(a);
+    tmp /= b;
+    return tmp;
+}
+
+template <class RhsValueType, class ValueType>
+EvaluationV<ValueType> operator*(const RhsValueType& a, const EvaluationV<ValueType>& b)
+{
+    EvaluationV<ValueType> result(b);
+    result *= a;
+    return result;
+}
+
+template <class ValueType>
+std::ostream& operator<<(std::ostream& os, const EvaluationV<ValueType>& eval)
+{
+    os << eval.value();
+    return os;
+}
+} } // namespace DenseAd, Opm
+
+// In Dune 2.3, the EvaluationV.hpp header must be included before the fmatrix.hh
+// header. Dune 2.4+ does not suffer from this because of some c++-foo.
```

Comment:
hm, since dune 2.3 support bit the dust a while ago, these hacks should probably be removed.